src/components/Form/Form.js

```diff
@@ -193,6 +193,8 @@ export function Form({ children, config }) {
         these arguments passed to the children function. */}
       {children({
         field,
+        errors: getFieldErrors(),
```

Comment:
I liked how this looked in Ethan's Formik example in terms of errors and values being property like  getterss

https://codesandbox.io/s/formik-example-tfqoi

Reply:
So I sort of hoped we'd avoid a situation like this where the client sees inside the Form abstraction, because i'm a bit worried that at some point a 'form engine' starts to lose its purpose. Most of the logic is still abstracted away though, such as updating state, validation, form validity, and well, everything else you got into `<Form>`. 

I like how they're labelled with getters too. i personally dislike that `getFieldErrors` is called and passed in as `errors`. I think it's unnecessary to call the getter on every render, and the "get" prefix does a good job of implying that you're getting something effectively read-only. Would you mind passing the getter directly without changing its name? (same for `getFieldValues`)

Reply:
I understand your stance on this, however, my dilemna on that is that Ethan's example has such tidy syntax that's easy to read:
```js
{values.b && !errors.b && (
                  <input...
```

Which would become:

```js
{getFieldValues().b && ! getFieldErrors().b && (
                  <input
```

And if you imagine a lot of JSX like that, it becomes unintelligible to my mind. And I'm pretty sure @ethantran would hate the later example and tell me, how come you're library is harder to use then Formik 😛 

Also, I think in fairness, we're already straying from things like getters, and  long names with `field` for the same reason—you could have called it `createField` or `newFieldFactory` or something, but it's so nice as just `field('name')` the way you did it; especially in declarative JSX land.

Reply:
I am ok with @kenning approach since the consumer can do this.
```js
{({ getFieldValues }) => {
      let values = getFieldValues()
      return {values.b ...}
}}
```

Reply:
Ok, if you're fine with that, then I'll use that—thanks for chiming in so quickly Ethan!

@kenning I'll do the `getFieldValues` and `getFieldErrors` thing now! :)

Reply:
Ok, so `{ !errors.field1 && values.field1 && field('field3')}` is interesting if you have not "form blurred" or really _form  touched_.

In terms of our current design, there's an interesting UX oddity, but it conforms to current rules.

1. `field1` has a constraint and doesn't allow more then 2 characters say
2. Upon typing first and second character, `field3` shows:

![image](https://user-images.githubusercontent.com/142403/67108017-6ee42a00-f182-11e9-95d3-6f612a3fe5f4.png)

3. But when you type a 3rd character, `field3` will be hidden as our `!errors.field1` has happened:

![image](https://user-images.githubusercontent.com/142403/67108082-9509ca00-f182-11e9-935f-d57afecd9d8a.png)

What's perhaps surprising is that although we've taken away the field3 because we have an error, the form hasn't been touched yet, so it's technically doing the correct thing. _Note, I don't consider clicking the button group turning the form into `touched` since it just felt crappy that the error hints kick in as soon as you go into an input which is where the `touched` (blured from input) experience feels the most useful/helpful._

I'm 🤔 maybe we live with this contrived issue and see if it shows up I dunno.

Reply:
The example looks good. In the future, maybe try exposing touched or interacted so `touched.x && value.x` or 'interacted.x && value.x`. See how the form behaves. Using `error` is still ok, but I think some users may find the second text input popping in and out too often just because the first input can become invalid. Although that may be just because the validator is super sensitive in your example.

Reply:
I also noticed that if you click field1 and then make first text input invalid with aaa and then click field2, you can't submit. I don't know the easiest way to fix this. That would require dynamic validation or validation of fields that depend on the state of other fields? or resetting fields when they are hidden?

Reply:
Yes, I saw this too. Do we need this for this week? I think I'll approach that when we get further down the line if not (or on spare cycles)